core/modules/user/user.module

```diff
@@ -1411,26 +1415,27 @@ function user_menu() {
     'menu_name' => 'user-menu',
   );
 
+  $user_login_appearance_setting = user_login_appearance_setting();
   $items['user/login'] = array(
     'title' => 'Log in',
     'page callback' => 'backdrop_get_form',
     'page arguments' => array('user_login'),
     'access callback' => 'user_is_anonymous',
-    'type' => MENU_DEFAULT_LOCAL_TASK,
+    'type' => ($user_login_appearance_setting == 'tabs') ? MENU_DEFAULT_LOCAL_TASK : MENU_CALLBACK,
```

Comment:
Instead of making this into `MENU_CALLBACK`, I suggest using `MENU_NORMAL_ITEM`. This will help us prevent the issue of the `user/login` being removed from the table `menu_links` when rebuilding the menus.

See my comment: https://github.com/backdrop/backdrop-issues/issues/6570#issuecomment-2561952164

MENU_NORMAL_ITEM has no side effect for this feature, and it helps prevent the issue I discovered.